diff -ur gcc-6.3.0-orig/gcc/config/gnu-user.h gcc-6.3.0/gcc/config/gnu-user.h
--- gcc-6.3.0-orig/gcc/config/gnu-user.h	2017-08-13 19:03:08.671572528 -0700
+++ gcc-6.3.0/gcc/config/gnu-user.h	2017-08-13 19:15:00.768588499 -0700
@@ -123,7 +123,7 @@
 
 #undef LINK_GCC_C_SEQUENCE_SPEC
 #define LINK_GCC_C_SEQUENCE_SPEC \
-  "%{static:--start-group} %G %L %{static:--end-group}%{!static:%G}"
+  "--start-group %G %L --end-group"
 
 /* Use --as-needed -lgcc_s for eh support.  */
 #ifdef HAVE_LD_AS_NEEDED
