diff --git a/pkg/urbit/vere/lmdb.c b/pkg/urbit/vere/lmdb.c
index b46cc66ee..b8f647dde 100644
--- a/pkg/urbit/vere/lmdb.c
+++ b/pkg/urbit/vere/lmdb.c
@@ -44,8 +44,8 @@ MDB_env* u3_lmdb_init(const char* log_path)
   // TODO: Start with forty gigabytes for the maximum event log size. We'll
   // need to do something more sophisticated for real in the long term, though.
   //
-  const size_t forty_gigabytes = 42949672960;
-  ret_w = mdb_env_set_mapsize(env, forty_gigabytes);
+  const size_t sixty_gigabytes = 64424509440;
+  ret_w = mdb_env_set_mapsize(env, sixty_gigabytes);
   if (ret_w != 0) {
     u3l_log("lmdb: failed to set database size: %s\n", mdb_strerror(ret_w));
     return 0;
