--- x86_64/qtdeclarative/examples/quick/tutorials/gettingStartedQml/filedialog/filedialog.pro.orig	2014-04-29 12:39:28.794200000 +0400
+++ x86_64/qtdeclarative/examples/quick/tutorials/gettingStartedQml/filedialog/filedialog.pro	2014-04-29 12:44:22.648600000 +0400
@@ -2,7 +2,7 @@
 CONFIG += plugin
 QT += qml
 
-DESTDIR +=  ../imports/FileDialog
+DESTDIR +=  ../imports/FileDialog/
 OBJECTS_DIR = tmp
 MOC_DIR = tmp
 
--- x86_64/qtdeclarative/examples/qml/tutorials/extending-qml/chapter6-plugins/import/import.pro.orig	2014-04-29 14:13:34.761600000 +0400
+++ x86_64/qtdeclarative/examples/qml/tutorials/extending-qml/chapter6-plugins/import/import.pro	2014-04-29 14:14:50.577600000 +0400
@@ -2,7 +2,7 @@
 CONFIG += plugin
 QT += qml quick
 
-DESTDIR = ../Charts
+DESTDIR = ../Charts/
 TARGET = $$qtLibraryTarget(chartsplugin)
 
 HEADERS += piechart.h \
--- x86_64/qtsensors/examples/sensors/grue/import/import.pro.orig	2014-04-29 12:38:40.995800000 +0400
+++ x86_64/qtsensors/examples/sensors/grue/import/import.pro	2014-04-29 12:43:12.823000000 +0400
@@ -2,7 +2,7 @@
 CONFIG += plugin
 
 TARGET = $$qtLibraryTarget(declarative_grue)
-DESTDIR = ../Grue
+DESTDIR = ../Grue/
 
 QT = core gui qml sensors
 
--- x86_64/qtsensors/examples/sensors/sensor_explorer/import/import.pro.orig	2014-04-29 12:38:17.143400000 +0400
+++ x86_64/qtsensors/examples/sensors/sensor_explorer/import/import.pro	2014-04-29 12:43:02.171000000 +0400
@@ -2,7 +2,7 @@
 CONFIG += plugin
 
 TARGET = $$qtLibraryTarget(declarative_explorer)
-DESTDIR = ../Explorer
+DESTDIR = ../Explorer/
 
 QT += qml sensors
 
