--- qt-everywhere-opensource-src-5.8.0/qtbase/mkspecs/features/configure_base.prf.orig	2014-04-20 21:21:48.808906400 +0100
+++ qt-everywhere-opensource-src-5.8.0/qtbase/mkspecs/features/configure_base.prf	2014-04-20 21:20:41.677381800 +0100
@@ -10,10 +10,10 @@
 !isEmpty(QMAKE_MAKE) {
     # We were called recursively. Use the same make.
 } else: if(equals(MAKEFILE_GENERATOR, UNIX)|equals(MAKEFILE_GENERATOR, MINGW)) {
-    !equals(QMAKE_HOST.os, Windows): \
-        QMAKE_MAKE = make
-    else: \
-        QMAKE_MAKE = mingw32-make
+    equals(QMAKE_HOST.os, Windows):isEmpty(QMAKE_SH): \
+        QMAKE_MAKE = mingw32-make
+    else: \
+        QMAKE_MAKE = make
 } else: if(equals(MAKEFILE_GENERATOR, MSVC.NET)|equals(MAKEFILE_GENERATOR, MSBUILD)) {
     QMAKE_MAKE = nmake
 } else {
