--- qt-everywhere-opensource-src-5.4.0.orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in.orig	2015-01-04 22:22:54.714460900 +0000
+++ qt-everywhere-opensource-src-5.4.0/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2015-01-04 22:34:11.801082600 +0000
@@ -283,7 +283,16 @@
 
 !!ENDIF // CMAKE_DEBUG_TYPE
 
-    file(GLOB pluginTargets \"${CMAKE_CURRENT_LIST_DIR}/Qt5$${CMAKE_MODULE_NAME}_*Plugin.cmake\")
+    file(GLOB pluginTargetsMaybe \"${CMAKE_CURRENT_LIST_DIR}/Qt5$${CMAKE_MODULE_NAME}_*.cmake\")
+    unset(pluginTargets)
+    if (pluginTargetsMaybe)
+        foreach(pluginTarget ${pluginTargetsMaybe})
+            file(STRINGS ${pluginTarget} matched REGEX Qt5$${CMAKE_MODULE_NAME}_PLUGINS)
+            if (matched)
+                list(APPEND pluginTargets ${pluginTarget})
+            endif()
+        endforeach()
+    endif()
 
     macro(_populate_$${CMAKE_MODULE_NAME}_plugin_properties Plugin Configuration PLUGIN_LOCATION)
         set_property(TARGET Qt5::${Plugin} APPEND PROPERTY IMPORTED_CONFIGURATIONS ${Configuration})
