diff -urN qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/features/default_post.prf.orig qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/features/default_post.prf
--- qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/features/default_post.prf.orig	2014-02-01 20:37:37.000000000 +0000
+++ qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/features/default_post.prf	2014-03-07 22:56:43.997614000 +0000
@@ -53,6 +53,7 @@
 }
 
 dll:win32: QMAKE_LFLAGS += $$QMAKE_LFLAGS_DLL
+static:win32: QMAKE_LFLAGS += $$QMAKE_LFLAGS_STATIC_LIB
 static:mac: QMAKE_LFLAGS += $$QMAKE_LFLAGS_STATIC_LIB
 staticlib:unix {
     QMAKE_CFLAGS += $$QMAKE_CFLAGS_STATIC_LIB
diff -urN qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/win32-g++/qmake.conf.orig qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/win32-g++/qmake.conf
--- qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/win32-g++/qmake.conf.orig	2014-02-01 20:37:37.000000000 +0000
+++ qt-everywhere-opensource-src-5.2.1/qtbase/mkspecs/win32-g++/qmake.conf	2014-03-07 22:55:45.900291100 +0000
@@ -74,6 +74,7 @@
 QMAKE_LFLAGS_CONSOLE    = -Wl,-subsystem,console
 QMAKE_LFLAGS_WINDOWS    = -Wl,-subsystem,windows
 QMAKE_LFLAGS_DLL        = -shared
+QMAKE_LFLAGS_STATIC_LIB += -static
 QMAKE_LFLAGS_CXX11      =
 QMAKE_LINK_OBJECT_MAX   = 10
 QMAKE_LINK_OBJECT_SCRIPT = object_script
