From 206e25be29ac115c84076a2b58bae3a6d0900eb1 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 18 Sep 2016 18:04:42 +0200
Subject: [PATCH 18/30] Use system zlib for build tools

---
 src/tools/bootstrap/bootstrap.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/bootstrap/bootstrap.pro b/src/tools/bootstrap/bootstrap.pro
index d51f9e98a4..2242b7c4f1 100644
--- a/src/tools/bootstrap/bootstrap.pro
+++ b/src/tools/bootstrap/bootstrap.pro
@@ -127,7 +127,7 @@ macx {
         ../../corelib/io/qstandardpaths_win.cpp
 }
 
-!qtConfig(system-zlib)|cross_compile {
+!qtConfig(system-zlib) {
     include(../../3rdparty/zlib.pri)
 } else {
     CONFIG += no_core_dep
-- 
2.11.1

