From 58f6e5d904464923eebcbb80ec0c774a30b30569 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 18 Sep 2016 18:50:21 +0200
Subject: [PATCH 23/30] Use correct pkg-config --static flag

---
 configure.pri | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.pri b/configure.pri
index 0c4b8ebbf8..8afb893e88 100644
--- a/configure.pri
+++ b/configure.pri
@@ -138,6 +138,9 @@ defineTest(qtConfTest_detectPkgConfig) {
             qtLog("Found pkg-config from path: $$pkgConfig")
         }
     }
+    !isEmpty(pkgConfig):qtConfig(static) {
+        pkgConfig = "$$pkgConfig --static"
+    }
 
     $$qtConfEvaluate("features.cross_compile") {
         # cross compiling, check that pkg-config is set up sanely
-- 
2.11.1

