diff -ur gcc-6.3.0-orig/gcc/config/linux.h gcc-6.3.0/gcc/config/linux.h
--- gcc-6.3.0-orig/gcc/config/linux.h	2017-07-24 08:32:44.412030567 -0700
+++ gcc-6.3.0/gcc/config/linux.h	2017-07-24 09:00:59.552111986 -0700
@@ -59,7 +59,7 @@
    the default.  */
 
 #define CHOOSE_DYNAMIC_LINKER1(LIBC1, LIBC2, LIBC3, LIBC4, LD1, LD2, LD3, LD4)	\
-  "%{" LIBC2 ":" LD2 ";:%{" LIBC3 ":" LD3 ";:%{" LIBC4 ":" LD4 ";:" LD1 "}}}"
+  "/dev/null"
 
 #if DEFAULT_LIBC == LIBC_GLIBC
 #define CHOOSE_DYNAMIC_LINKER(G, U, B, M) \
