diff -ur cfe-3.9.1.src.orig/lib/Driver/ToolChains.cpp cfe-3.9.1.src/lib/Driver/ToolChains.cpp
--- cfe-3.9.1.src.orig/lib/Driver/ToolChains.cpp	2016-11-23 13:25:50.000000000 -0800
+++ cfe-3.9.1.src/lib/Driver/ToolChains.cpp	2017-09-10 15:56:04.298686786 -0700
@@ -1406,18 +1406,6 @@
 
     // Then look for gcc installed alongside clang.
     Prefixes.push_back(D.InstalledDir + "/..");
-
-    // Then look for distribution supplied gcc installations.
-    if (D.SysRoot.empty()) {
-      // Look for RHEL devtoolsets.
-      Prefixes.push_back("/opt/rh/devtoolset-4/root/usr");
-      Prefixes.push_back("/opt/rh/devtoolset-3/root/usr");
-      Prefixes.push_back("/opt/rh/devtoolset-2/root/usr");
-      Prefixes.push_back("/opt/rh/devtoolset-1.1/root/usr");
-      Prefixes.push_back("/opt/rh/devtoolset-1.0/root/usr");
-      // And finally in /usr.
-      Prefixes.push_back("/usr");
-    }
   }
 
   // Loop over the various components which exist and select the best GCC
Only in cfe-3.9.1.src/lib/Frontend: #InitHeaderSearch.cpp#
