diff -ur ld64-274.2-orig/src/ld/ld.cpp ld64-274.2/src/ld/ld.cpp
--- ld64-274.2-orig/src/ld/ld.cpp	2017-09-29 07:44:04.316338766 -0700
+++ ld64-274.2/src/ld/ld.cpp	2017-09-29 20:28:44.609155692 -0700
@@ -21,12 +21,6 @@
  *
  * @APPLE_LICENSE_HEADER_END@
  */
- 
-// start temp HACK for cross builds
-extern "C" double log2 ( double );
-//#define __MATH__
-// end temp HACK for cross builds
-
 
 #include <stdlib.h>
 #include <sys/types.h>
diff -ur ld64-274.2-orig/src/ld/Options.h ld64-274.2/src/ld/Options.h
--- ld64-274.2-orig/src/ld/Options.h	2017-09-29 07:44:04.319672117 -0700
+++ ld64-274.2/src/ld/Options.h	2017-09-29 20:37:11.961284911 -0700
@@ -502,8 +502,9 @@
 		bool					containsNonWildcard(const char*) const;
 		bool					empty() const			{ return fRegular.empty() && fWildCard.empty(); }
 		bool					hasWildCards() const	{ return !fWildCard.empty(); }
-		NameSet::iterator		regularBegin() const	{ return fRegular.begin(); }
-		NameSet::iterator		regularEnd() const		{ return fRegular.end(); }
+		// thanks to tpoechtrager's cctools-port for this fix
+		NameSet::const_iterator	regularBegin() const	{ return fRegular.begin(); }
+		NameSet::const_iterator	regularEnd() const		{ return fRegular.end(); }
 		void					remove(const NameSet&); 
 		std::vector<const char*>		data() const;
 	private:
diff -ur ld64-274.2-orig/src/ld/Snapshot.h ld64-274.2/src/ld/Snapshot.h
--- ld64-274.2-orig/src/ld/Snapshot.h	2017-09-29 07:44:04.319672117 -0700
+++ ld64-274.2/src/ld/Snapshot.h	2017-09-29 20:32:25.044874041 -0700
@@ -88,7 +88,6 @@
 
     friend class SnapshotArchiveFileLog;
     
-    typedef std::vector<void(^)(void)> SnapshotLog;    
 
     struct strcompclass {
         bool operator() (const char *a, const char *b) const { return ::strcmp(a, b) < 0; }
@@ -128,7 +127,6 @@
     void addFrameworkArg(const char *framework);
     void addDylibArg(const char *dylib);
 
-    SnapshotLog fLog;           // log of events that recorded data in a snapshot prior to createSnapshot()
     bool fRecordArgs;           // record command line 
     bool fRecordObjects;        // record .o files 
     bool fRecordDylibSymbols;   // record referenced dylib/framework symbols
