diff -ur cctools-port-c1cc758/cctools/include/foreign/extern.h cctools-port-patched/cctools/include/foreign/extern.h
--- cctools-port-c1cc758/cctools/include/foreign/extern.h	2017-10-01 13:47:04.000000000 -0700
+++ cctools-port-patched/cctools/include/foreign/extern.h	2017-10-29 09:01:17.731958699 -0700
@@ -1 +1,2 @@
+#pragma once
 #define __private_extern__ __attribute__((visibility("hidden")))
diff -ur cctools-port-c1cc758/cctools/include/mach-o/dyld.h cctools-port-patched/cctools/include/mach-o/dyld.h
--- cctools-port-c1cc758/cctools/include/mach-o/dyld.h	2017-10-01 13:47:04.000000000 -0700
+++ cctools-port-patched/cctools/include/mach-o/dyld.h	2017-10-29 09:00:59.738428248 -0700
@@ -27,9 +27,7 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#if defined(__MWERKS__) && !defined(__private_extern__)
-#define __private_extern__ __declspec(private_extern)
-#endif
+#include <extern.h>
 
 #include <mach-o/loader.h>
 #include <AvailabilityMacros.h>
