diff -ur devcon-orig/devcon.h devcon/devcon.h
--- devcon-orig/devcon.h	2017-04-03 22:32:04.995677665 -0700
+++ devcon/devcon.h	2017-04-03 22:49:54.215677665 -0700
@@ -25,6 +25,7 @@
 #include <newdev.h>
 #include <objbase.h>
 #include <strsafe.h>
+#include <driverspecs.h>
 #include <io.h>
 #include <fcntl.h>
 
diff -ur devcon-orig/dump.cpp devcon/dump.cpp
--- devcon-orig/dump.cpp	2017-04-03 22:32:04.995677665 -0700
+++ devcon/dump.cpp	2017-04-03 22:54:09.635677664 -0700
@@ -407,7 +407,7 @@
     return TRUE;
 }
 
-
+__stdcall
 UINT DumpDeviceDriversCallback(_In_ PVOID Context, _In_ UINT Notification, _In_ UINT_PTR Param1, _In_ UINT_PTR Param2)
 /*++
 
