diff -ur devcon-orig/devcon.h devcon/devcon.h
--- devcon-orig/devcon.h	2017-04-30 07:21:34.058007997 -0700
+++ devcon/devcon.h	2017-04-30 07:36:15.028007999 -0700
@@ -25,6 +25,7 @@
 #include <newdev.h>
 #include <objbase.h>
 #include <strsafe.h>
+#include <driverspecs.h>
 #include <io.h>
 #include <fcntl.h>
 
diff -ur devcon-orig/dump.cpp devcon/dump.cpp
--- devcon-orig/dump.cpp	2017-04-30 07:21:34.058007997 -0700
+++ devcon/dump.cpp	2017-04-30 07:23:19.708007999 -0700
@@ -407,8 +407,7 @@
     return TRUE;
 }
 
-
-UINT DumpDeviceDriversCallback(_In_ PVOID Context, _In_ UINT Notification, _In_ UINT_PTR Param1, _In_ UINT_PTR Param2)
+UINT CALLBACK DumpDeviceDriversCallback(_In_ PVOID Context, _In_ UINT Notification, _In_ UINT_PTR Param1, _In_ UINT_PTR Param2)
 /*++
 
 Routine Description:
