--- qt-5.8.0-orig/src/platformsupport/linuxaccessibility/dbusconnection.cpp
+++ qt-5.8.0/src/platformsupport/linuxaccessibility/dbusconnection.cpp
@@ -75,7 +75,7 @@
     connect(dbusWatcher, SIGNAL(serviceRegistered(QString)), this, SLOT(serviceRegistered()));
 
     // If it is registered already, setup a11y right away
-    if (c.interface()->isServiceRegistered(A11Y_SERVICE))
+    if (c.interface() && c.interface()->isServiceRegistered(A11Y_SERVICE))
         serviceRegistered();
 
     // In addition try if there is an xatom exposing the bus address, this allows applications run as root to work
--- qtbase-5.8.0-orig/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
+++ qtbase-5.8.0/src/platformsupport/themes/genericunix/qgenericunixthemes.cpp
@@ -122,7 +122,7 @@
 {
     QDBusConnection connection = QDBusConnection::sessionBus();
     QString registrarService = QStringLiteral("com.canonical.AppMenu.Registrar");
-    return connection.interface()->isServiceRegistered(registrarService);
+    return connection.interface() && connection.interface()->isServiceRegistered(registrarService);
 }
 
 static bool isDBusGlobalMenuAvailable()
