From 65c70e8cad7d6c234b00a4d3240a98cae49465e7 Mon Sep 17 00:00:00 2001
From: David Grayson <davidegrayson@gmail.com>
Date: Wed, 10 May 2017 07:07:04 -0700
Subject: [PATCH 2/2] specstrings.h: Add an include guard.

---
 mingw-w64-headers/include/specstrings.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mingw-w64-headers/include/specstrings.h b/mingw-w64-headers/include/specstrings.h
index 984dd64b..bcb9fc3c 100644
--- a/mingw-w64-headers/include/specstrings.h
+++ b/mingw-w64-headers/include/specstrings.h
@@ -3,6 +3,10 @@
  * This file is part of the mingw-w64 runtime package.
  * No warranty is given; refer to the file DISCLAIMER.PD within this package.
  */
+
+#ifndef SPECSTRINGS_H
+#define SPECSTRINGS_H
+
 #define __specstrings
 
 #include <sal.h>
@@ -330,3 +334,5 @@ extern "C" {
 #endif
 
 #include <driverspecs.h>
+
+#endif
-- 
2.12.1

