diff -ur qtbase-opensource-src-5.9.2-orig/mkspecs/common/linux.conf qtbase-opensource-src-5.9.2/mkspecs/common/linux.conf
--- qtbase-opensource-src-5.9.2-orig/mkspecs/common/linux.conf	2017-10-26 08:10:12.922646692 -0700
+++ qtbase-opensource-src-5.9.2/mkspecs/common/linux.conf	2017-10-26 21:44:37.695088447 -0700
@@ -28,7 +28,7 @@
 
 QMAKE_LIBS              =
 QMAKE_LIBS_DYNLOAD      = -ldl
-QMAKE_LIBS_X11          = -lXext -lX11 -lm
+QMAKE_LIBS_X11          = -lXext -lX11 -lxcb -lXau
 QMAKE_LIBS_EGL          = -lEGL
 QMAKE_LIBS_OPENGL       = -lGL
 QMAKE_LIBS_OPENGL_ES2   = -lGLESv2
