diff -ur qtbase-everywhere-src-5.11.2-orig/mkspecs/common/g++-win32.conf qtbase-everywhere-src-5.11.2/mkspecs/common/g++-win32.conf
--- qtbase-everywhere-src-5.11.2-orig/mkspecs/common/g++-win32.conf	2018-10-23 21:31:56.864905432 -0700
+++ qtbase-everywhere-src-5.11.2/mkspecs/common/g++-win32.conf	2018-10-23 21:43:31.797114564 -0700
@@ -48,10 +48,8 @@
 QMAKE_LFLAGS_WINDOWS    = -Wl,-subsystem,windows
 QMAKE_LFLAGS_DLL        = -shared
 QMAKE_LFLAGS_GCSECTIONS = -Wl,--gc-sections
-equals(QMAKE_HOST.os, Windows) {
-    QMAKE_LINK_OBJECT_MAX = 10
-    QMAKE_LINK_OBJECT_SCRIPT = object_script
-}
+QMAKE_LINK_OBJECT_MAX = 10
+QMAKE_LINK_OBJECT_SCRIPT = object_script
 QMAKE_EXT_OBJ           = .o
 QMAKE_EXT_RES           = _res.o
 QMAKE_PREFIX_SHLIB      =
