--- trunk/gcc/ubsan.c	2016/09/03 09:20:03	239970
+++ trunk/gcc/ubsan.c	2016/09/03 10:57:05	239971
@@ -1469,7 +1469,7 @@
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 
