diff -ur libusbp-1.0.1/install_helper/install_helper_windows.c libusbp-1.0.1-patched/install_helper/install_helper_windows.c
--- libusbp-1.0.1/install_helper/install_helper_windows.c	2016-03-20 15:04:00.000000000 -0700
+++ libusbp-1.0.1-patched/install_helper/install_helper_windows.c	2017-01-20 19:17:26.124674666 -0800
@@ -3,7 +3,7 @@
  * rundll32. */
 
 #include <windows.h>
-#include <MsiQuery.h>
+#include <msiquery.h>
 #include <setupapi.h>
 #include <strsafe.h>
 
