diff -ur gcc-5.4.0/gcc/c/c-convert.c gcc-5.4.0-patched/gcc/c/c-convert.c
--- gcc-5.4.0/gcc/c/c-convert.c	2015-03-09 04:57:33.000000000 -0700
+++ gcc-5.4.0-patched/gcc/c/c-convert.c	2017-01-13 23:25:37.858012332 -0800
@@ -87,7 +87,7 @@
   if ((invalid_conv_diag
        = targetm.invalid_conversion (TREE_TYPE (expr), type)))
     {
-      error (invalid_conv_diag);
+      error ("%s", invalid_conv_diag);
       return error_mark_node;
     }
 
diff -ur gcc-5.4.0/gcc/c/c-decl.c gcc-5.4.0-patched/gcc/c/c-decl.c
--- gcc-5.4.0/gcc/c/c-decl.c	2016-04-18 08:21:52.000000000 -0700
+++ gcc-5.4.0-patched/gcc/c/c-decl.c	2017-01-13 22:12:29.734359331 -0800
@@ -6110,7 +6110,7 @@
 	    errmsg = targetm.invalid_return_type (type);
 	    if (errmsg)
 	      {
-		error (errmsg);
+		error ("%s", errmsg);
 		type = integer_type_node;
 	      }
 
@@ -6918,7 +6918,7 @@
 	  errmsg = targetm.invalid_parameter_type (type);
 	  if (errmsg)
 	    {
-	      error (errmsg);
+	      error ("%s", errmsg);
 	      TREE_VALUE (typelt) = error_mark_node;
 	      TREE_TYPE (parm) = error_mark_node;
 	      arg_types = NULL_TREE;
diff -ur gcc-5.4.0/gcc/c/c-typeck.c gcc-5.4.0-patched/gcc/c/c-typeck.c
--- gcc-5.4.0/gcc/c/c-typeck.c	2016-02-17 05:10:59.000000000 -0800
+++ gcc-5.4.0-patched/gcc/c/c-typeck.c	2017-01-13 22:20:24.697692665 -0800
@@ -3364,7 +3364,7 @@
       else if ((invalid_func_diag =
 		targetm.calls.invalid_arg_for_unprototyped_fn (typelist, fundecl, val)))
 	{
-	  error (invalid_func_diag);
+	  error ("%s", invalid_func_diag);
 	  return -1;
 	}
       else
@@ -3848,7 +3848,7 @@
   if ((invalid_op_diag
        = targetm.invalid_unary_op (code, TREE_TYPE (xarg))))
     {
-      error_at (location, invalid_op_diag);
+      error_at (location, "%s", invalid_op_diag);
       return error_mark_node;
     }
 
@@ -5620,7 +5620,7 @@
   char *ofwhat;
 
   /* The gmsgid may be a format string with %< and %>. */
-  error_at (loc, gmsgid);
+  error_at (loc, "%s", gmsgid);
   ofwhat = print_spelling ((char *) alloca (spelling_length () + 1));
   if (*ofwhat)
     inform (loc, "(near initialization for %qs)", ofwhat);
@@ -5638,7 +5638,7 @@
   bool warned;
 
   /* The gmsgid may be a format string with %< and %>. */
-  warned = pedwarn (location, opt, gmsgid);
+  warned = pedwarn (location, opt, "%s", gmsgid);
   ofwhat = print_spelling ((char *) alloca (spelling_length () + 1));
   if (*ofwhat && warned)
     inform (location, "(near initialization for %qs)", ofwhat);
@@ -5657,7 +5657,7 @@
   bool warned;
 
   /* The gmsgid may be a format string with %< and %>. */
-  warned = warning_at (loc, opt, gmsgid);
+  warned = warning_at (loc, opt, "%s", gmsgid);
   ofwhat = print_spelling ((char *) alloca (spelling_length () + 1));
   if (*ofwhat && warned)
     inform (loc, "(near initialization for %qs)", ofwhat);
@@ -10373,7 +10373,7 @@
   if ((invalid_op_diag
        = targetm.invalid_binary_op (code, type0, type1)))
     {
-      error_at (location, invalid_op_diag);
+      error_at (location, "%s", invalid_op_diag);
       return error_mark_node;
     }
 
diff -ur gcc-5.4.0/gcc/c-family/c-common.c gcc-5.4.0-patched/gcc/c-family/c-common.c
--- gcc-5.4.0/gcc/c-family/c-common.c	2016-03-30 05:29:47.000000000 -0700
+++ gcc-5.4.0-patched/gcc/c-family/c-common.c	2017-01-13 23:48:08.788012332 -0800
@@ -9941,11 +9941,11 @@
       message = NULL;
     }
   else
-    error (gmsgid);
+    error ("%s", gmsgid);
 
   if (message)
     {
-      error (message);
+      error ("%s", message);
       free (message);
     }
 #undef catenate_messages
diff -ur gcc-5.4.0/gcc/c-family/cilk.c gcc-5.4.0-patched/gcc/c-family/cilk.c
--- gcc-5.4.0/gcc/c-family/cilk.c	2015-01-21 13:23:04.000000000 -0800
+++ gcc-5.4.0-patched/gcc/c-family/cilk.c	2017-01-13 23:57:14.061345666 -0800
@@ -1395,13 +1395,13 @@
   if (contains_array_notation_expr (expr))
     {
       loc = get_error_location (expr, loc);
-      error_at (loc, array_msgid);
+      error_at (loc, "%s", array_msgid);
       return true;
     }
   if (walk_tree (&expr, contains_cilk_spawn_stmt_walker, NULL, NULL))
     {
       loc = get_error_location (expr, loc);
-      error_at (loc, spawn_msgid);
+      error_at (loc, "%s", spawn_msgid);
       return true;
     }
   return false;
diff -ur gcc-5.4.0/gcc/fold-const.c gcc-5.4.0-patched/gcc/fold-const.c
--- gcc-5.4.0/gcc/fold-const.c	2016-05-23 05:51:29.000000000 -0700
+++ gcc-5.4.0-patched/gcc/fold-const.c	2017-01-14 00:06:34.841345666 -0800
@@ -335,7 +335,7 @@
 	}
     }
   else if (issue_strict_overflow_warning (wc))
-    warning (OPT_Wstrict_overflow, gmsgid);
+    warning (OPT_Wstrict_overflow, "%s", gmsgid);
 }
 
 /* Return true if the built-in mathematical function specified by CODE
diff -ur gcc-5.4.0/gcc/gimple-ssa-isolate-paths.c gcc-5.4.0-patched/gcc/gimple-ssa-isolate-paths.c
--- gcc-5.4.0/gcc/gimple-ssa-isolate-paths.c	2015-09-21 02:55:57.000000000 -0700
+++ gcc-5.4.0-patched/gcc/gimple-ssa-isolate-paths.c	2017-01-14 00:18:54.474678999 -0800
@@ -451,7 +451,7 @@
 				 "local variable");
 
 		      if (warning_at (gimple_location (stmt),
-				      OPT_Wreturn_local_addr, msg))
+			             OPT_Wreturn_local_addr, "%s", msg))
 			inform (DECL_SOURCE_LOCATION(valbase), "declared here");
 		      tree zero = build_zero_cst (TREE_TYPE (val));
 		      gimple_return_set_retval (return_stmt, zero);
diff -ur gcc-5.4.0/libcpp/expr.c gcc-5.4.0-patched/libcpp/expr.c
--- gcc-5.4.0/libcpp/expr.c	2015-01-16 09:40:15.000000000 -0800
+++ gcc-5.4.0-patched/libcpp/expr.c	2017-01-13 21:53:32.137692664 -0800
@@ -678,10 +678,10 @@
 
 	  if (CPP_OPTION (pfile, c99))
             cpp_warning_with_line (pfile, CPP_W_LONG_LONG, virtual_location,
-				   0, message);
+                                  0, "%s", message);
           else
             cpp_pedwarning_with_line (pfile, CPP_W_LONG_LONG,
-				      virtual_location, 0, message);
+                                     virtual_location, 0, "%s", message);
         }
 
       result |= CPP_N_INTEGER;
diff -ur gcc-5.4.0/libcpp/macro.c gcc-5.4.0-patched/libcpp/macro.c
--- gcc-5.4.0/libcpp/macro.c	2016-03-30 05:43:44.000000000 -0700
+++ gcc-5.4.0-patched/libcpp/macro.c	2017-01-13 22:02:47.714359333 -0800
@@ -3084,7 +3084,7 @@
 	     function-like macros, but not at the end.  */
 	  if (following_paste_op)
 	    {
-	      cpp_error (pfile, CPP_DL_ERROR, paste_op_error_msg);
+              cpp_error (pfile, CPP_DL_ERROR, "%s", paste_op_error_msg);
 	      return false;
 	    }
 	  break;
@@ -3097,7 +3097,7 @@
 	     function-like macros, but not at the beginning.  */
 	  if (macro->count == 1)
 	    {
-	      cpp_error (pfile, CPP_DL_ERROR, paste_op_error_msg);
+              cpp_error (pfile, CPP_DL_ERROR, "%s", paste_op_error_msg);
 	      return false;
 	    }
 
