--- gcc-r265561/gcc/calls.c
+++ gcc/gcc/calls.c
@@ -4738,6 +4738,8 @@
      what we need.  */
   if (crtl->preferred_stack_boundary < PREFERRED_STACK_BOUNDARY)
     crtl->preferred_stack_boundary = PREFERRED_STACK_BOUNDARY;
+  if (crtl->stack_alignment_needed < crtl->preferred_stack_boundary)
+    crtl->stack_alignment_needed = crtl->preferred_stack_boundary;
 
   /* If this kind of value comes back in memory,
      decide where in memory it should come back.  */
