--- gcc-7-branch/gcc/cfgexpand.c
+++ gcc-7-branch/gcc/cfgexpand.c
@@ -6466,6 +6466,14 @@
   find_many_sub_basic_blocks (blocks);
   purge_all_dead_edges ();

+  /* After initial rtl generation, call back to finish generating
+     exception support code.  We need to do this before cleaning up
+     the CFG as the code does not expect dead landing pads.  */
+  if (fun->eh->region_tree != NULL)
+    finish_eh_generation ();
+
+  /* Call expand_stack_alignment after finishing all
+     updates to crtl->preferred_stack_boundary.  */
   expand_stack_alignment ();

   /* Fixup REG_EQUIV notes in the prologue if there are tailcalls in this
@@ -6473,12 +6481,6 @@
   if (crtl->tail_call_emit)
     fixup_tail_calls ();

-  /* After initial rtl generation, call back to finish generating
-     exception support code.  We need to do this before cleaning up
-     the CFG as the code does not expect dead landing pads.  */
-  if (fun->eh->region_tree != NULL)
-    finish_eh_generation ();
-
   /* Remove unreachable blocks, otherwise we cannot compute dominators
      which are needed for loop state verification.  As a side-effect
      this also compacts blocks.
