diff -ur angle-src-orig/src/angle.gyp angle-src/src/angle.gyp
--- angle-src-orig/src/angle.gyp	2017-02-07 07:42:35.090343332 -0800
+++ angle-src/src/angle.gyp	2017-02-09 08:44:52.752006998 -0800
@@ -286,7 +286,7 @@
                 }
             ]
         }],
-        ['OS=="win"',
+        ['OS=="win" and 0',
         {
             'targets':
             [
diff -ur angle-src-orig/src/libGLESv2.gypi angle-src/src/libGLESv2.gypi
--- angle-src-orig/src/libGLESv2.gypi	2017-02-07 07:42:35.113676666 -0800
+++ angle-src/src/libGLESv2.gypi	2017-02-09 09:19:37.985340331 -0800
@@ -1078,7 +1078,7 @@
                         '<@(libangle_null_sources)',
                     ],
                 }],
-                ['angle_build_winrt==0 and OS=="win"',
+                ['angle_build_winrt==0 and OS=="win" and 0',
                 {
                     'dependencies':
                     [
