diff -ur angle-src-orig/gyp/common_defines.gypi angle-src/gyp/common_defines.gypi
--- angle-src-orig/gyp/common_defines.gypi	2017-02-07 07:42:35.087009999 -0800
+++ angle-src/gyp/common_defines.gypi	2017-02-09 07:29:32.245340333 -0800
@@ -44,6 +44,9 @@
     ],
     'conditions':
     [
+        ['OS=="win"', {
+            'defines': [ 'MINGW_HAS_SECURE_API=1', ],
+        }],
         ['use_ozone==1',
         {
             'defines': [ 'USE_OZONE' ],
diff -ur angle-src-orig/src/angle.gyp angle-src/src/angle.gyp
--- angle-src-orig/src/angle.gyp	2017-02-07 07:42:35.090343332 -0800
+++ angle-src/src/angle.gyp	2017-02-09 08:44:52.752006998 -0800
@@ -286,7 +286,7 @@
                 }
             ]
         }],
-        ['OS=="win"',
+        ['OS=="win" and 0',
         {
             'targets':
             [
diff -ur angle-src-orig/src/common/string_utils.cpp angle-src/src/common/string_utils.cpp
--- angle-src-orig/src/common/string_utils.cpp	2017-02-07 07:42:35.093676666 -0800
+++ angle-src/src/common/string_utils.cpp	2017-02-09 07:38:28.762006999 -0800
@@ -11,7 +11,9 @@
 
 #include <algorithm>
 #include <stdlib.h>
-#include <string.h>
+#ifndef __MINGW32__
+#include <sestring.h>
+#endif
 #include <fstream>
 #include <sstream>
 
diff -ur angle-src-orig/src/libGLESv2.gypi angle-src/src/libGLESv2.gypi
--- angle-src-orig/src/libGLESv2.gypi	2017-02-07 07:42:35.113676666 -0800
+++ angle-src/src/libGLESv2.gypi	2017-02-09 09:19:37.985340331 -0800
@@ -1078,7 +1078,7 @@
                         '<@(libangle_null_sources)',
                     ],
                 }],
-                ['angle_build_winrt==0 and OS=="win"',
+                ['angle_build_winrt==0 and OS=="win" and 0',
                 {
                     'dependencies':
                     [
