diff -ur gdb-7.12.1-orig/gdb/source.c gdb-7.12.1/gdb/source.c
--- gdb-7.12.1-orig/gdb/source.c	2017-02-24 19:33:13.340349333 -0800
+++ gdb-7.12.1/gdb/source.c	2017-02-24 19:34:40.660349333 -0800
@@ -1103,10 +1103,7 @@
 	}
     }
 
-  if (IS_ABSOLUTE_PATH (filename))
     {
-      /* If filename is absolute path, try the source path
-	 substitution on it.  */
       char *rewritten_filename = rewrite_source_path (filename);
 
       if (rewritten_filename != NULL)
