diff -ur qtbase-opensource-src-5.9.2-orig/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm qtbase-opensource-src-5.9.2/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm
--- qtbase-opensource-src-5.9.2-orig/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm	2017-11-03 20:37:01.268207968 -0700
+++ qtbase-opensource-src-5.9.2/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm	2017-11-03 20:37:14.031628522 -0700
@@ -830,7 +830,7 @@
 
 QFixed QCoreTextFontEngine::emSquareSize() const
 {
-    return QFixed::QFixed(int(CTFontGetUnitsPerEm(ctfont)));
+    return QFixed(int(CTFontGetUnitsPerEm(ctfont)));
 }
 
 QFontEngine *QCoreTextFontEngine::cloneWithSize(qreal pixelSize) const
